### ISSUE TITLE
Set up automatic unit tests on build

### ISSUE THREAD

**Issue by cwn-dev** [strasnote/strasnote#20]

**Comment by bencgreen** [comment]
This can be done as follows:

Tests.csproj (in root)

```xml
<Project Sdk="Microsoft.Build.Traversal/2.0.24">
	<ItemGroup>
		<ProjectReference Include="Tests/**/*.csproj" />
	</ItemGroup>
</Project>
```

GitHub action:

```yml
name: Test

on:
  push:
    branches-ignore:
      - 'main'
    tags-ignore:
      - '**'
    paths:
      - '.github/workflows/test.yml'
      - 'Tests/**'

jobs:
  restore-build-test:
    runs-on: ubuntu-latest
    steps:
    - 
      name: Prepare - Checkout Code
      uses: actions/checkout@v2
    - 
      name: Prepare - Setup .NET
      uses: actions/setup-dotnet@v1
      with:
        dotnet-version: '5.0.x'
    - 
      name: Build and Run tests
      run: |
        dotnet restore Test.csproj
        dotnet build Test.csproj --no-restore -c Release
        dotnet test Test.csproj --no-restore --no-build -c Release --verbosity normal
    - 
      if: success()
      name: Finish - Success
      ... do something ...
    - 
      if: failure()
      name: Finish - Failure
      ... do something ...
```

**Comment by bencgreen** [comment]
Closed by #21
